packages/bitcore-node/src/routes/api/stats.ts

```diff
@@ -1,22 +1,42 @@
 import { Request, Response } from 'express';
 import { ChainStateProvider } from '../../providers/chain-state';
 import { SetCache, CacheTimes } from '../middleware';
+
 const router = require('express').Router({ mergeParams: true });
 
 router.get('/', async function(_: Request, res: Response) {
   return res.send(404);
 });
 
+let cache = {};
+let updating = false;
+
 router.get('/daily-transactions', async function(req: Request, res: Response) {
   const { chain, network } = req.params;
+  const cacheKey = chain + ':' + network;
+  const updateCache = async () => {
+    const hasFreshData = cache[cacheKey] && cache[cacheKey].expiry > Date.now();
+    if (!updating && !hasFreshData) {
+      updating = true;
+      try {
+        let dailyTxs = await ChainStateProvider.getDailyTransactions({
+          chain,
+          network,
+          startDate: '',
+          endDate: ''
+        });
+        cache[cacheKey] = { dailyTxs, expiry: Date.now() + CacheTimes.Day };
+        updating = false;
+        return cache[cacheKey];
+      } catch (e) {
+        updating = false;
+        throw e;
+      }
+    }
+  };
   try {
-    let dailyTxs = await ChainStateProvider.getDailyTransactions({
-      chain,
-      network,
-      startDate: req.query.startDate,
-      endDate: req.query.endDate
-    });
-    SetCache(res, CacheTimes.Day);
+    const { dailyTxs } = await updateCache();
+    SetCache(res, CacheTimes.Day, CacheTimes.Hour);
     return res.json(dailyTxs);
   } catch (err) {
     return res.status(500).send(err);
```

Comment:
why we are sending the err back to the client?
should we just print it to the log and return "Internal server error" message or something?

It seems to me we are disclosing to much information on a production server.